staging/src/k8s.io/kubectl/pkg/cmd/create/create_namespace.go

```diff
@@ -55,40 +84,108 @@ func NewCmdCreateNamespace(f cmdutil.Factory, ioStreams genericclioptions.IOStre
 		Long:                  namespaceLong,
 		Example:               namespaceExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(options.Complete(f, cmd, args))
-			cmdutil.CheckErr(options.Run())
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
```

Comment:
Based on the code walk through [video](https://www.youtube.com/watch?v=sQTATR_X_sU)(Check out the video starting from 43.30) usually it has Complete, Validate, and Run

Reply:
TBH I don't thing actually Validate() is necessary, because this is a simple create that have no validation other then if the name is present.

And this is validated in https://github.com/kubernetes/kubernetes/pull/96556/files#diff-29d11e54ae61c7c6d756e3c18ca0629990dfd51dcd46922b6af683d3c90347f0R105

As an example, take a look in create_priorityclass.go which is a really simple object.

OTOH, create_serviceaccount have a validation that does the same validation of  NameFromCommandArgs.

So, I guess here we can skip the Validate() if this is not necessary :)

Reply:
Let's do this way: let's keep Validate() as a pattern, even if this validates twice (on Complete and on Validate) the same thing, and then we can remove it/cleanup later

Reply:
Perhaps we need a more  complicated validation here.As described in the [url](https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace),the name of a namespace must be a valid DNS label which is described [here](https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace).But it  conflicts with previous [code](https://github.com/kubernetes/kubernetes/blob/1119a505aca14467accedf850daf30aa9c532ef2/staging/src/k8s.io/kubectl/pkg/generate/versioned/namespace.go#L75)

Reply:
hum, but we can improve the code. I guess anyway who returns an error if the name is or is not  DNS formated is the APIServer.

Let me test your PR here and see how things happen :)

Reply:
Following up with the test:

```
_output/bin/kubectl create ns lalala###123
The Namespace "lalala###123" is invalid: metadata.name: Invalid value: "lalala###123": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```

Because all the objects are validated against the validation.IsDNS1123Label() function (from api-machinery) I guess this is the normal behavior and we don't need to take care of this (for now, but maybe an improvement in the future validating this before sending to the APIServer and making also dry-run complain, wdyt @soltysh @pwittrock ?)

So for now what you can do is keep the Validate() as is (porting the code from the generator), and removing the call of Validate() from createNamespace() :)